modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesKernelBoot.java

```diff
@@ -171,19 +181,29 @@ public static void setVirtualHost(VirtualHost defaultVirtualHost) {
     /**
      * This will trigger graceful shutdown of andes broker
      *
-     * @throws Exception
+     * @throws AndesException
      */
-    public static void shutDownAndesKernel() throws Exception {
+    public static void shutDownAndesKernel() throws AndesException {
+
+        // Set flag so independent threads can act accordingly
+        isKernelShuttingDown = true;
```

Comment:
Shutdown listener is not intended to invoke any custom logic before or after shutting down the andes kernel. (apart from unergistering the QpidService from osgi.) So setting the value here is timely.